api/docker-entrypoint.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+#include suntcions file
+source functions.sh
+set -e
+WAIT_THRESHOLD="${WAIT_THRESHOLD:-20}"
+SLEEP_SECONDS="${SLEEP_SECONDS:-3}"
+
+info()
+{
+    echo "Info: $*"
+}
+
+warn()
+{
+    echo "Warning: $*"
+}
+
+fatal()
+{
+    echo "Fatal: $*"
+}
+
+is_setup_needed()
+{
+  ! is_existing_user 'medic-api'
+}
+
+is_existing_user()
+{
+  local user="$1"
+  local url="`get_couchdb_url`"
+  local id="org.couchdb.user:$user"
+  shift 1
+
+  local userdoc=$(curl -X GET $url/_users/$id | jq '._id?')
+
+  if [ "$userdoc" = \"org.couchdb.user:$user\" ]; then
+    if [ ! -f /srv/storage/$user/passwd/$user ]; then
+      create_couchdb_admin "$user"
+      return 1
+    fi
+    return 0
+  else
+    return 1
+  fi
+}
+
+create_couchdb_put()
+{
+  local should_auth="$1"
+  local payload="$2"
+  local payload_is_file="$3"
+
+  local cmd='cat'
+
+  if [ -z "$payload_is_file" ]; then
+    cmd='echo'
+  fi
+
+  if [ "$#" -gt 1 ]; then
+    echo -n 'data = "\"' &&
+    "$cmd" "$payload" &&
+    echo '\""'
+  fi
+}
+
+
+
+perform_couchdb_lockdown()
+{
+  local url="`get_couchdb_url`"
+  curl -X PUT $url/_node/couchdb@$COUCH_NODE_NAME/_config/chttpd/require_valid_user -d '"true"'
+}
+
+create_system_databases()
+{
+  local url="`get_couchdb_url`" &&
+  for db in _users _replicator _global_changes; do
+    if ! curl -sfX PUT "$url/$db" >/dev/null; then
+      warn "Failed to created system database '$db'"
+      return "$?"
+    fi
+  done
+}
+
+postinstall()
+{
+  info 'Creating system databases' &&
+  create_system_databases \
+    || fatal 'Failed to create one or more system databases'
+
+  info 'Restricting CouchDB to authorized users only' &&
+  perform_couchdb_lockdown
+
+  if [ "$?" -ne 0 ]; then
+    fatal "Failed to lock out invalid CouchDB users"
+  fi
+
+  info 'Creating CouchDB service accounts'
+
+  create_couchdb_admin 'medic-api'
```

Comment:
I think we should convert this to JavaScript, so we can have full dev support and complete unit tests. 
It's place can be either in the `server-checks` shared lib or another shared lib. 
Since `sentinel` will need to create its own account too, we double win because we reduce code duplication.

Reply:
I can help with the JavaScript part, if you'd like!

Reply:
Thanks! I am re-writing the entrypoint  functions in javascript and only leave installations in the dockerfile.